cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -177,7 +177,10 @@ func (s *storageExt) Start(_ context.Context, host component.Host) error {
 		var metricsFactory storage.MetricsFactory
 		var err error
 		if cfg.Prometheus != nil {
-			metricsFactory, err = prometheus.NewFactoryWithConfig(*cfg.Prometheus, s.telset.Logger)
+			metricsFactory, err = prometheus.NewFactoryWithConfig(
+				*cfg.Prometheus,
+				getMetricsFactory(metricStorageName, "prometheus"),
```

Comment:
@yurishkuro Are we okay with publishing the metrics under the `jaeger_storage` namespace to keep it consistent with the backend stores? We're currently publishing them under `jaeger_metricstore`.

Reply:
lgtm

Reply:
however, perhaps we could also add the `role` attribute here: `tracestore`, `metricstore`.

Reply:
@yurishkuro wouldn't this get overridden because the `role` attribute is being used to differentiate between the primary and archive factories for some storage implementations?

Reply:
But that only applies to tracestores, does it not?

Reply:
yeah - so should we only add it for `metricstore`?

Reply:
I would add it for both. If the downstream function overrides the attribute, so be it. Once we remove ArchiveStorageFactory, this might change.